### PR TITLE
Collection settings ES6 conversion and tests

### PR DESCRIPTION
**Pull request by srallen**
I decided to do some ES6 conversions and test writing for collection settings since we don't have the bandwidth to do some of the collection feature implementation at the moment.

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] Did you deploy a staging branch?
- [ ] Did you get any type checking errors from [Babel Typecheck](https://github.com/codemix/babel-plugin-typecheck)

## Optional

- [x] If it's a new component, is it in ES6? Is it clear of warnings from ESLint?
- [x] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?

### CODE REVIEW

File: app/collections/settings.jsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    this.deleteCollection = this.deleteCollection.bind(this);
+    this.handleToggle = this.handleToggle.bind(this);
+    this.redirect = this.redirect.bind(this);
+  }
+
+  componentDidMount() {
+    this.props.collection.listen('delete', this.redirect);
+  }
+
+  componentDidUnMount() {
```
Comment (wgranger) [inline app/collections/settings.jsx]: Is this now a valid lifecycle method?

File: app/collections/settings.jsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+              <button className="major-button" disabled={true}><i className="fa fa-spinner" /></button>
+              {' '}
+            </div>}
+
+          {!this.state.isDeleting &&
+            <div>
+              <button className="major-button" onClick={handleDelete}>Yes, delete it!</button>
+              {' '}
+              <button className="minor-button" onClick={resolve}>No, don't delete it.</button>
+            </div>}
```
Comment (eatyourgreens) [inline app/collections/settings.jsx]: One of these buttons should have `autoFocus={true}`.

File: app/collections/settings.jsx (hunk 1 above)
Comment (srallen) [inline app/collections/settings.jsx]: Yep it's a typo on my part.

File: app/collections/settings.jsx (hunk 2 above)
Comment (srallen) [inline app/collections/settings.jsx]: Added it to the cancel since that made sense to me. But could add it to the confirm button instead.

**Comment by srallen** [comment]
Addressed comments and cleaned up some linter stuff that wasn't configured correctly.

File: app/collections/settings.jsx (hunk 2 above)
Comment (eatyourgreens) [inline app/collections/settings.jsx]: Cancel makes sense to me. Confirm runs the risk of accidentally pressing Enter and deleting the collection.

Maybe reverse the order of the buttons, so you can tab from cancel to confirm? Sorry to be a pain about this.

Comment (srallen) [inline app/collections/settings.jsx]: Not a big deal to reverse. Makes sense.

**Comment by srallen** [comment]
Staged at https://collection-settings.pfe-preview.zooniverse.org/

**Comment by wgranger** [comment]
LGTM

**Pull request merged**